postgres-pubsub.js

```diff
@@ -29,23 +29,44 @@ class PostgresPubSub extends PubSub {
     this.connected = false;
   }
 
-  async connect() {
-    console.log('calling connect')
-    // confusingly, `pgListen.connect()` will reject if the first connection attempt fails
-    // but then it will retry and emit a `connected` event if it later connects
-    // see https://github.com/andywer/pg-listen/issues/32
-    // so we put logic on the `connected` event
-    this.pgListen.events.on('connected', async () => {
-      await Promise.all(this.triggers.map((eventName) => {
-        return this.pgListen.listenTo(eventName);
-      }));
-      this.connected = true;
+  /**
+   * @returns
+   * Rejects when any of the following occur:
+   *   1. pg-listen's initial `connect` fails for an exotic (i.e., non-ECONNREFUSED)
+   *      reason.
+   *   2. pg-listen emits 'error', likely indicating initial connection failed
+   *      even after repeated attempts.
+   *   3. Connection to the database was successful, but at least one
+   *      `LISTEN` query failed.
+   *
+   * Fulfills otherwise, indicating all of the requested triggers are now being
+   * listened to.
+   */
+  connect() {
+    return new Promise(async (resolve, reject) => {
```

Comment:
A wise reviewer (@bendrucker) once told me that if you find yourself using the `Promise` constructor, in almost all cases it can be avoided. Promises needed to be created from scratch at the very base of a codebase, like if you were writing a library like lodash, but in most cases, if you're dealing with promises it's because they're been passed up to you. In that case, you can usually return them.

Can you find a way to keep `connect` as an `async` function, and still achieve the same functionality?

Reply:
Hrm, yeah, my first few attempts were actually `Promise.all`-based, which wasn't a great fit. `Promise.race` might work well, if I can codify the 4 possible fates of the promise that this function returns (as described in its docstring) in terms of 4 promises fed into `Promise.race`.

Will give it a shot. No guarantees that it'll be _simpler_ than this, though. 😬

Reply:
Okeydoke, updated. Let me know if that feels better. Oh and have a great weekend 😄

Reply:
I looked into it today and starting from your proposal here realized it's difficult to get around the event emitter without using the promise constructor >.< I don't think I was appreciating the difficulty of that in your first proposal.
I do like the `Promise.race` better now.

I'm submitting a PR to your PR where I did a quick refactor that made it a little easier to read for me. If you think that works, merge it, and then we'll merge this one :)

Thanks again for the help!

Reply:
Nvm, Github changed their UI. I'll merge this one and have you as a reviewer on my refactor.